pkg/cmd/server/apis/config/validation/master.go

```diff
@@ -171,6 +174,12 @@ func ValidateMasterConfig(config *configapi.MasterConfig, fldPath *field.Path) V
 func ValidateMasterAuthConfig(config configapi.MasterAuthConfig, fldPath *field.Path) ValidationResults {
 	validationResults := ValidationResults{}
 
+	if len(config.OAuthMetadataFile) > 0 {
+		if _, _, err := oauthutil.LoadOAuthMetadataFile(config.OAuthMetadataFile); err != nil {
```

Comment:
This check has made the oauthmetadatafile format part of our API.  Make a TODO to fix the package dependency order and promote the struct.

Reply:
Added in discovery.go, I have a card to deal with this as a followup